test/test_env.py

```diff
@@ -0,0 +1,15 @@
+"""Contains tests for creating and initializing custom Julia projects"""
+
+import unittest
+from pysr import julia_helpers
+
+
+class TestJuliaProject(unittest.TestCase):
+    def test_custom_shared_env(self):
+        """Test that we can use PySR in a custom shared env"""
+        test_env_name = "@pysr_test_env"
+        julia_helpers.install(julia_project=test_env_name, quiet=True)
+        Main = julia_helpers.init_julia(julia_project=test_env_name)
+        Main.eval("using SymbolicRegression")
+        # TODO: Test that we are actually in the correct env.
+        # TODO: Delete the env at the end.
```

Comment:
Not sure if there is any way to do this? I suppose I need to create a temp env instead (but that's a different question).

Reply:
You could `pushfirst!` a temp directory into `DEPOT_PATH`. Then the shared environment would be created in the temp directory. Then you can `pop!` the tempdir off `DEPOT_PATH` and recursively delete it.

Reply:
If the `DEPOT_PATH` changes, and you activate a new environment, will it correctly update? It seems like it is not switching to the new setting for some reason (see error)

```python
Traceback (most recent call last):
  File "/Users/mcranmer/Documents/PySR/test/test_env.py", line 24, in test_custom_shared_env
    self.assertEqual(cur_project_dir, potential_shared_project_dirs)
AssertionError: '/Users/mcranmer/.julia/environments' != '/var/folders/1h/xyppkvx52cl6w3_h8bw_gdqh0000gr/T/tmp8dknp59s/environments'
```

i.e., the current project is still in `/Users/mcranmer/.julia/environments` rather than the new depot